### PR TITLE
mate-base/mate-settings-daemon: Fix incompatible function pointer types

### PR DESCRIPTION
**Pull request by listout**
Closes: https://bugs.gentoo.org/881315

### CODE REVIEW

File: mate-base/mate-settings-daemon/files/mate-settings-daemon-1.26.0-fix-buid-with-clang16.patch [hunk 1]
```diff
@@ -0,0 +1,14 @@
+Bug: https://bugs.gentoo.org/881315
```
Comment (thesamesam) [inline mate-base/mate-settings-daemon/files/mate-settings-daemon-1.26.0-fix-buid-with-clang16.patch:1]: You've sent it upstream, I assume? If so, where's the link? If not, why not? 😉 

Always upstream!

File: mate-base/mate-settings-daemon/files/mate-settings-daemon-1.26.0-fix-buid-with-clang16.patch [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+Author: David Carlos Manuelda <StormByte@gmail.com>
+--- a/plugins/a11y-keyboard/msd-a11y-keyboard-atspi.c
++++ b/plugins/a11y-keyboard/msd-a11y-keyboard-atspi.c
+@@ -128,7 +128,7 @@
+         /* init AT-SPI if needed */
+         atspi_init ();
+ 
+-        self->listener = atspi_device_listener_new (on_key_press_event,
++        self->listener = atspi_device_listener_new ((AtspiDeviceListenerCB)on_key_press_event,
```
Comment (thesamesam) [inline mate-base/mate-settings-daemon/files/mate-settings-daemon-1.26.0-fix-buid-with-clang16.patch:10]: I'm not yet sure if this is actually right. I need to check.

Comment (listout) [reply mate-base/mate-settings-daemon/files/mate-settings-daemon-1.26.0-fix-buid-with-clang16.patch:10]: My initial patch was to remove const from `on_key_press_event`'s paramer
```diff
 static gboolean
-on_key_press_event (const AtspiDeviceEvent *event,
+on_key_press_event (AtspiDeviceEvent *event,
                     void                   *user_data G_GNUC_UNUSED)
 {
         /* don't ring on capslock itself, that's taken care of by togglekeys
```

Comment (thesamesam) [reply mate-base/mate-settings-daemon/files/mate-settings-daemon-1.26.0-fix-buid-with-clang16.patch:10]: Thanks for asking - I think the patch as it is (the cast) is better now I looked at it harder.

**Comment by thesamesam** [comment]
Thanks. This is a good example of why:
1. upstreaming is important;
2. casting should _not_ be the first thing one does if we can help it

**Comment by listout** [comment]
I'll update the patch to match that of maintainers.

**Comment by listout** [comment]
Huh, in the upstream correct PR by dev, they are just removing the const parameter from the function, https://github.com/mate-desktop/mate-settings-daemon/pull/400/commits/42e91f2805c508317b4d26166cb3db332dd313e8.

**Comment by thesamesam** [comment]
Yeah, you can see they had to think for a while about whether it's the best way to go though and that it's a trade-off (and the API changed).

**Review by juippis — APPROVED** [review]

**Pull request closed without merging**